miden-lib/asm/miden/sat/internal/prologue.masm

```diff
@@ -679,78 +679,99 @@ end
 proc.process_consumed_notes_data
     # load the consumed notes data onto the advice stack
     exec.layout::get_nullifier_com adv.push_mapval dropw
+    # => [...]
 
     # read the number of consumed notes from the advice provider
     adv_push.1
+    # => [num_notes, ...]
 
     # store the number of consumed notes
     dup exec.layout::set_total_num_consumed_notes
+    # => [num_notes, ...]
 
-    # assert the number of consumed notes is within limits
-    dup exec.constants::get_max_num_consumed_notes lte assert
+    # assert the number of consumed notes is within limits; since max number of consumed notes is
+    # smaller than 2^32, we can use a more efficient u32 comparison
+    dup exec.constants::get_max_num_consumed_notes u32assert2 u32lte assert
+    # => [num_notes, ...]
 
     # loop over consumed notes and read data
     # ---------------------------------------------------------------------------------------------
 
     # initialize counter for consumed notes
     push.0
+    # => [num_processed_notes, num_notes, ...]
 
     # check if the number of consumed notes is greater then 0. Conditional for the while loop.
     dup.1 dup.1 neq
+    # => [has_more_notes, num_processed_notes, num_notes, ...]
 
     # loop and read note data from the advice provider
     while.true
         dup exec.process_consumed_note
+        # => [num_processed_notes, num_notes, ...]
 
         # increment consumed note counter and check if we should loop again
         add.1 dup.1 dup.1 neq
+        # => [has_more_notes, num_processed_notes + 1, num_notes, ...]
     end
 
     # drop counter
     drop
+    # => [num_notes, ...]
 
     # compute nullifier commitment
     # ---------------------------------------------------------------------------------------------
 
     # initiate counter for nullifiers
     push.0
+    # => [num_processed_notes, num_notes, ...]
```

Comment:
can you call this counter `num_nullifiers` or so?

Reply:
We are still technically iterating over notes (and just extracting a nullifier from a note at each iteration) - so, I'd keep it as is.